phoenix-core/src/main/java/org/apache/phoenix/schema/json/PhoenixJson.java

```diff
@@ -0,0 +1,255 @@
+            }
+            return phoenixJson;
+        } catch (NumberFormatException nfe) {
+            throw new SQLException("path: " + Arrays.asList(paths) + " not found.", nfe);
+        }
+    }
+
+    /**
+     * Get {@link PhoenixJson} for a given json paths. For example :
+     * <p>
+     * <code>
+     * {"f2":{"f3":1},"f4":{"f5":99,"f6":{"f7":"2"}}}'
+     * </code>
+     * <p>
+     * for this source json, if we want to know the json at path {'f4','f6'} it will return
+     * {@link PhoenixJson} object for json {"f7":"2"}. It always returns the last key if same key
+     * exist more than once.
+     * <p>
+     * If the given path is unreachable then it return null.
+     * @param paths {@link String []} of path in the same order as they appear in json.
+     * @return {@link PhoenixJson} for the json against @paths.
+     */
+    public PhoenixJson getPhoenixJsonOrNull(String[] paths) {
+        try {
+            return getPhoenixJsonInternal(paths);
+        } catch (NumberFormatException nfe) {
+            // ignore
+        }
+        return null;
+    }
+
+    /**
+     * Serialize the current {@link PhoenixJson} to String. Its required for
+     * json_extract_path_text(). If we just return node.toString() it will wrap String value in
+     * double quote which is not the expectation, hence avoiding calling toString() on
+     * {@link JsonNode} until PhoenixJson represent a Json Array or container for Json object. If
+     * PhoenixJson just represent a {@link ValueNode} then it should return value returned from
+     * objects toString().
+     */
+    public String serializeToString() {
+        if (this.rootNode == null || this.rootNode.isNull()) {
+            return null;
+        } else if (this.rootNode.isValueNode()) {
+
+            if (this.rootNode.isNumber()) {
+                return this.rootNode.getNumberValue().toString();
+            } else if (this.rootNode.isBoolean()) {
+                return String.valueOf(this.rootNode.getBooleanValue());
+            } else if (this.rootNode.isTextual()) {
+                return this.rootNode.getTextValue();
+            } else {
+                return getJsonAsString();
+            }
+        } else if (this.rootNode.isArray()) {
+            return getJsonAsString();
+        } else if (this.rootNode.isContainerNode()) {
+            return getJsonAsString();
+        }
+
+        return null;
+
+    }
+
+    @Override
+    public String toString() {
+        return getJsonAsString();
+    }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((this.rootNode == null) ? 0 : this.rootNode.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        PhoenixJson other = (PhoenixJson) obj;
```

Comment:
I think String match will faster since JsonNode.equals() traverse the entire Tree Model of Json.